src/server/api/blockallowlist.ts

```diff
@@ -1,9 +1,22 @@
 import { APIConfig, FastifyTypebox } from '.'
 import { Type } from '@sinclair/typebox'
 
+import createError from 'http-errors'
+
 import Store from '../store/index.js'
 import ActivityPubSystem from '../apsystem.js'
 
+export function validateBlocklistFormat (mention: string): void {
+  const sections = mention.split('@')
+  const correctSections = sections.length === 3
+  const startsEmpty = sections[0] === ''
+  const hasUserAndDomain = (sections[1] !== '') && (sections[2] !== '')
```

Comment:
blocking a whole instance is `@*@doma.in` no?

Reply:
Yup! In that cause username is set to `*`. `@@domain.com` would be invalid for example.